RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToClusterAlgo.h

```diff
@@ -0,0 +1,151 @@
+                      SOA_SCALAR(uint8_t, maxSequentialBad),
+                      SOA_SCALAR(uint8_t, maxAdjacentBad),
+                      SOA_SCALAR(float, minGoodCharge),
+                      SOA_SCALAR(uint16_t, clusterSizeLimit))
+
+  using StripClustersAuxSoA = StripClustersAuxSoALayout<>;
+  using StripClustersAuxView = StripClustersAuxSoA::View;
+  using StripClustersAuxConstView = StripClustersAuxSoA::ConstView;
+  using StripClustersAuxHost = PortableHostCollection<StripClustersAuxSoA>;
+}  // namespace sistrip
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  using namespace ::sistrip;
+  using StripClustersAuxDevice = PortableCollection<StripClustersAuxSoA>;
+}  // namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip
+
+ASSERT_DEVICE_MATCHES_HOST_COLLECTION(sistrip::StripClustersAuxDevice, sistrip::StripClustersAuxHost);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  struct FEDChMetadata {
+    uint32_t detId;
+    uint16_t fedId;
+    uint8_t fedCh;
+    //
+    uint16_t fedChOfs;
+    uint32_t fedChOfs_wrt_rawFedId;
+    uint8_t fedBufferRO;
+  };
+
+  class PortableFEDMover {
+  public:
+    PortableFEDMover(Queue& queue, uint32_t rawBufferSize, uint32_t fedChannelsNb)
+        : buffer_(cms::alpakatools::make_host_buffer<uint8_t[]>(queue, rawBufferSize)),
+          mapping_(fedChannelsNb, queue),
+          bufferSize_(0),
+          channelNb_(0),
+          fedChannelNb_(fedChannelsNb),
+          offset4FedId_(sistrip::FED_ID_MAX + 1) {}
+
+    void fillBuffer(const std::vector<const FEDRawData*>& raw) {
+      for (uint16_t fedId = sistrip::FED_ID_MIN; fedId <= sistrip::FED_ID_MAX; ++fedId) {
+        if (raw[fedId]) {
+          std::memcpy(buffer_.data() + bufferSize_, raw[fedId]->data(), raw[fedId]->size());
+          offset4FedId_[fedId] = bufferSize_;
+          bufferSize_ += raw[fedId]->size();
+        }
+      }
+    }
+
+    void fillMapping(const std::vector<FEDChMetadata>& channelsMeta) {
+      for (uint32_t i = 0; i < fedChannelNb_; i++, channelNb_++) {
+        const auto& channel = channelsMeta[i];
+        mapping_->detID(i) = channel.detId;
+        mapping_->fedID(i) = channel.fedId;
+        mapping_->fedCh(i) = channel.fedCh;
+        mapping_->fedChOfs(i) = channel.fedChOfs;
+        mapping_->fedChDataOfsBuf(i) = channel.fedChOfs_wrt_rawFedId + offset4FedId_[channel.fedId];
+        mapping_->readoutMode(i) = channel.fedBufferRO;
+      }
+    }
+
+    auto buffer() { return buffer_; }
+    uint32_t bufferSize() const { return bufferSize_; }
+
+    auto mapping() { return std::move(mapping_); }
+    uint32_t channelNb() const { return channelNb_; }
+
+  private:
+    cms::alpakatools::host_buffer<uint8_t[]> buffer_;
+    SiStripMappingHost mapping_;
+
+    uint32_t bufferSize_;
+    uint32_t channelNb_;
+    uint32_t fedChannelNb_;
```

Comment:
aren't `channelNb_` and `fedChannelNb_` always going to have the same value after the call to `fillMapping()` ?

would it make sense to drop one of the two ?